Modules/Core/Common/include/itkInPlaceImageFilter.h

```diff
@@ -147,7 +147,15 @@ class ITK_TEMPLATE_EXPORT InPlaceImageFilter : public ImageToImageFilter<TInputI
   void
   AllocateOutputs() override
   {
-    this->InternalAllocateOutputs(IsSame<TInputImage, TOutputImage>());
+    if (std::is_same<TInputImage, TOutputImage>::value)
```

Comment:
Looks like a candidate for C++17 compile-time evaluated `if constexpr` 😸

Reply:
Even today, constant propagation should take care of optimization. If you have a TODO list of things for C++17, you could add this to it 😄